src/assistants/openai/openai_api_version.rs

```diff
@@ -0,0 +1,175 @@
+use anyhow::{anyhow, Result};
+use reqwest::header::{self, HeaderMap, HeaderValue};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::str::FromStr;
+
+use crate::constants::OPENAI_API_URL;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Eq, PartialEq)]
+pub enum OpenAIAssistantVersion {
+    V1,
+    V2,
+    Azure,
+}
+
+impl OpenAIAssistantVersion {
+    pub(crate) fn get_endpoint(&self, resource: &OpenAIAssistantResource) -> String {
```

Comment:
This is the most important change of the PR. This method abstracts the endpoint that are called for different OpenAI APIs. Instead of manual assembly of urls it introduces an enum to manage that. It also abstracts away the differences between OpenAI API and Azure APIs.